lib/usb/usb_standard.c

```diff
@@ -46,7 +46,10 @@ int usbd_register_set_config_callback(usbd_device *usbd_dev,
 
 	for (i = 0; i < MAX_USER_SET_CONFIG_CALLBACK; i++) {
 		if (usbd_dev->user_callback_set_config[i]) {
-			continue;
+			if (usbd_dev->user_callback_set_config[i] != callback)
+				continue;
+			else
+				return 0;
```

Comment:
I would write this differently
```
if ((usbd_dev->user_callback_set_config[i] != NULL) &&
   (usbd_dev->user_callback_set_config[i] == callback)) {
    return 0;
}
```
The multiple continues in there are a bit weird.

Reply:
it's so that it continues properly with the right return codes though.

Reply:
You are right @karlp here is (for me) the more readable and understandable code:
```c
int usbd_register_set_config_callback(usbd_device *usbd_dev,
				       usbd_set_config_callback callback)
{
	int i;
	for (i = 0; i < MAX_USER_SET_CONFIG_CALLBACK; i++) {
		if ((usbd_dev->user_callback_set_config[i] == NULL) ||
                    (usbd_dev->user_callback_set_config[i] == callback)) {
			usbd_dev->user_callback_set_config[i] = callback;
                        return 0;
               }
        }
       return -1;
}
```

Reply:
hehe :)  I'd worked up this version myself locally: 
```
	for (i = 0; i < MAX_USER_SET_CONFIG_CALLBACK; i++) {
		if (usbd_dev->user_callback_set_config[i]) {
			if (usbd_dev->user_callback_set_config[i] == callback) {
				return 0;
			}
			continue;
		}

		usbd_dev->user_callback_set_config[i] = callback;
		return 0;
	}
```